lib/install/validate/network.go

```diff
@@ -74,40 +82,157 @@ func (v *Validator) checkNetworkConflict(bridgeNetName, otherNetName, otherNetTy
 	}
 }
 
+// portGroupConfig gets the input config for all networks
+// for use in checking that the config is valid
+func (v *Validator) portGroupConfig(input *data.Data, counts map[string]int, ips map[string][]data.NetworkConfig) {
+	defer trace.End(trace.Begin(""))
+
+	if input.ManagementNetwork.Name != "" {
+		counts[input.ManagementNetwork.Name]++
+		if !ip.Empty(input.ManagementNetwork.IP) {
+			ips[input.ManagementNetwork.Name] = append(ips[input.ManagementNetwork.Name], input.ManagementNetwork)
+		}
+	}
+	if input.ClientNetwork.Name != "" {
+		counts[input.ClientNetwork.Name]++
+		if !ip.Empty(input.ClientNetwork.IP) {
+			ips[input.ClientNetwork.Name] = append(ips[input.ClientNetwork.Name], input.ClientNetwork)
+		}
+	}
+	if input.ExternalNetwork.Name != "" {
+		counts[input.ExternalNetwork.Name]++
+		if !ip.Empty(input.ExternalNetwork.IP) {
+			ips[input.ExternalNetwork.Name] = append(ips[input.ExternalNetwork.Name], input.ExternalNetwork)
+		}
+	}
+}
+
+// checkPortGroups checks that network config is valid
+// prevent assigning > 1 static IP to the same port group
+func (v *Validator) checkPortGroups(counts map[string]int, ips map[string][]data.NetworkConfig) error {
+	defer trace.End(trace.Begin(""))
+
+	for pg := range ips {
```

Comment:
As a forward looking note, we should be able to handle multiple IPs on one interface. This could be leveraged by customers to get separation between client & other networks without having separate port groups.

This is related again to source routing support.